rust/worker/src/index/hnsw_provider.rs

```diff
@@ -355,6 +355,8 @@ impl HnswIndexProvider {
             match res {
                 Ok(_) => {
                     println!("Flushed hnsw index file: {}", file);
+                    std::fs::remove_file(file_path)
```

Comment:
note (to myself mostly) - think about the retry story in case of a partial flush

Reply:
I thought about only deleting the files at the end of the flush to require that all writes were successful, the tradeoff is that can result in dangling temp files if the upload partially fails